c10/core/DispatchKeySet.h

```diff
@@ -34,6 +34,7 @@ namespace c10 {
 class DispatchKeySet final {
 public:
   enum Full { FULL };
```

Comment:
Thinking about it, I probably would have put those into static factory functions instead of constructors with an enum argument, i.e.

    static DispatchKeySet DispatchKeySet::full();
    static DispatchKeySet DispatchKeySet::raw();
    static DispatchKeySet DispatchKeySet::full_after(...);

Seems more standard. But not a strong opinion if you wanna keep it as is.

Reply:
I'm in the habit of doing constructors because it makes field initialization syntax better (doesn't matter here as we have copy/move constructors but when you don't have those it makes a ton of difference.)